### PR TITLE
Macro update and build fix

### PR DESCRIPTION
**Pull request by crop2000**
in my pr to build_faust i added cargo::rerun things to the build function this lead to to many reruns i think ... so i think it should be removed.
the other commit is to add a functionality to control flags in faust_macro via the faust declare block.

### CODE REVIEW

File: faust-build/src/lib.rs [hunk 1]
```diff
@@ -155,29 +153,27 @@ impl FaustBuilder {
...
-                    .to_camel_case()
-            }
-        };
+        let struct_name = self.get_struct_name();
 
         output
             .arg("-lang")
             .arg("rust")
             .arg("-xml")
-            .arg("--output-dir")
```
Comment (obsoleszenz) [inline faust-build/src/lib.rs:179]: Why did you remove this?

File: faust-state/src/lib.rs [hunk 2]
```diff
@@ -452,22 +439,21 @@ impl ParamsBuilder {
...
         let idx = idx.0;
-        if self.inner.contains_key(&idx) {
-            let node = self.inner.get_mut(&idx).unwrap();
-            node.label = label.to_string();
-            node.typ = typ;
-            if let Some(mut metadata) = metadata {
-                node.metadata.append(metadata.as_mut());
-            }
-        } else {
+        if let std::collections::hash_map::Entry::Vacant(e) = self.inner.entry(idx) {
```
Comment (obsoleszenz) [inline faust-state/src/lib.rs:442]: Oh nice! Didn't know about this

File: faust-build/src/lib.rs (hunk 1 above)
Comment (crop2000) [reply faust-build/src/lib.rs:179]: i think it makes more sense to either rely on faust to do what it does by default. Or to specify a path with filename (like we do for the generated code) the output-dir is a third case right in the middle, so i would rather not provide this possible interface.

File: faust-state/src/lib.rs (hunk 2 above)
Comment (crop2000) [reply faust-state/src/lib.rs:442]: this is one of those things clippy knows for us :)

Comment (obsoleszenz) [reply faust-state/src/lib.rs:442]: clippy is such a good boy!
